src/lib.rs

```diff
@@ -945,24 +946,31 @@ fn fill_todo(
             let dirs = fs::read_dir(path).and_then(|d| {
                 d.map(|e| {
                     e.map(|e| {
-                        let path = if curdir {
-                            PathBuf::from(e.path().file_name().unwrap())
+                        // We sort by filename a few lines below. However, it is actually quite
+                        // expensive to extract the filename from PathBuf. Since we already know the
+                        // filename from `DirEntry` here, we store it proactively (we still
+                        // have to heap allocate it).
+                        // Then we use this cached filename for sorting, and further on work only
+                        // with the full path.
+                        let (path, filename) = if curdir {
+                            (PathBuf::from(e.path().file_name().unwrap()), e.file_name())
                         } else {
-                            e.path()
+                            (e.path(), e.file_name())
                         };
-                        PathWrapper::from_dir_entry(path, e)
+                        (PathWrapper::from_dir_entry(path, e), filename)
                     })
                 })
-                .collect::<Result<Vec<_>, _>>()
+                .collect::<Result<Vec<(PathWrapper, OsString)>, _>>()
             });
             match dirs {
                 Ok(mut children) => {
                     if options.require_literal_leading_dot {
-                        children
-                            .retain(|x| !x.file_name().unwrap().to_str().unwrap().starts_with('.'));
+                        children.retain(|x| {
+                            !x.0.file_name().unwrap().to_str().unwrap().starts_with('.')
```

Comment:
you could use the filename here too

Reply:
I wasn't 100% sure if it has the same behavior, tbh. In other words, is `DirEntry::file_name()` guaranteed to return the same thing as `DirEntry::path().file_name()`? In particular around normalizing the leading `.`.